### PR TITLE
auth-impovement

### PR DESCRIPTION
**Pull request by mprudnik**
- added fastify/request-context (async-local-storage)
 - moved session storage to context instead of request mixin

### CODE REVIEW

File: src/server/server.js [hunk 1]
```diff
@@ -84,7 +92,9 @@ export const init = async (router, infra, config) => {
           const payload = schemaDefinition.source
             ? req[schemaDefinition.source]
             : {};
-          const session = authDefinition ? req.session : undefined;
+          const session = authDefinition
+            ? server.requestContext.get('session')
```
Comment (georgolden) [inline src/server/server.js:96]: `requestContext.get` returns undefined by default. Why you check for authDefinition here?

File: src/server/server.js [hunk 2]
```diff
@@ -84,7 +92,9 @@ export const init = async (router, infra, config) => {
...
+            : undefined;
           const result = await bus.command(command, payload, session);
```
Comment (georgolden) [inline src/server/server.js:98]: Why do you pass a session to a command? If there is actual data, that the service needs to know, then give it to the payload. In your particular case, it is just userId - pass it with payload.
In other cases, the user sessions can be combined with the user's cache and have permissions, fingerprints, etc.

**Review by georgolden — CHANGES_REQUESTED** [review]
1. I suggest using Redis for sessions. It can scale and it is less tricky, cause you just need to fetch data by token after authentication and make authorization easier.
Generally, for the monolith, it is ok to store session data in the server's memory, but it is not scalable, and with node js request scopes is not simpler than just using Redis.
2. Fastify has a plugin that will make code less messy. https://github.com/fastify/fastify-auth

**Comment by mprudnik** [comment]
@georgolden 
> 1. I suggest using Redis for sessions. It can scale and it is less tricky, cause you just need to fetch data by token after authentication and make authorization easier.
>    Generally, for the monolith, it is ok to store session data in the server's memory, but it is not scalable, and with node js request scopes is not simpler than just using Redis.
> 2. Fastify has a plugin that will make code less messy. https://github.com/fastify/fastify-auth

1. I'm not storing session data in memory - it is stored in Postgres (at least for now), changing it to Redis won't change the code a lot. The code in this PR updates how I check the authentication for HTTP request/WS connection and pass its data to the service layer.
Would it be more convenient to pass session data as command metadata?
For example:
```
const data = req.body;
const session = server.requestContext.get('session');
const result = await bus.command('service.method', { meta: { session }, data });
```
I don't want to mix session and request data since if command requires authenticated access but not uses session data it will result in:
```
const someService = {
  someMethod: async ({ data: { param1, param2, userId: _u, ...rest } }) => { ...
}
```
Havind it inside `meta` allow to use it if needed. `({ meta: { session }, data: { param1, ...rest } }) => { ...`

**Pull request closed without merging**